### ISSUE TITLE
Model Selection FourCastNet

### ISSUE THREAD

**Issue by thomasy666** [cyber-shuttle/cybershuttle-ai4science-foundry#4]
## Model Selection
FourCastNet
## Research Area
This is a model developed by NVIDIA that uses neural networks to predict future weather.
## Motivation
I first learned about weather prediction models when I learned how Markov chains functioned. I am also interested in how neural networks function in inference. This model takes a 0.25-degree lat/long area and then applies 10+ factors, including temperature and wind, into an adaptive Fourier neural operator to make its prediction. I am interested in how this is faster than conventional weather models and how its accuracy differs. Additionally, I'm interested in how the Fourier Domain works mathematically. 
## Semester Plan
The plan for this semester is to be able to deploy this model on the GT computing cluster successfully. I will produce detailed instructions for others to be able to follow the same steps and deploy the model as well. I will also learn the science behind the model. This includes understanding the input and how it converts the parameters. The second step would be to use this model on a downstream task. For this specific model, I will pick a grid globally and predict the weather forecast for the next 24/48 hours, and compare its results to the real-world results for accuracy. This will then have a peer production aspect as well. 
At the end of the semester, I will train the model myself with a limited-sized dataset and learn the process of training and the impacts of a limited training set. I will also look at factors such as learning curves and analysis, and try to make this reproducible as well.
## Resources
There are numerous resources regarding this topic
1. Github: https://github.com/NVlabs/FourCastNet
2. Huggingface: https://huggingface.co/nvidia/fourcastnet1
3. NVIDIA API: https://build.nvidia.com/nvidia/fourcastnet/deploy
4. ERA5 Data: https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels?tab=overview
5. Documentation: https://docs.nvidia.com/physicsnemo/latest/physicsnemo/examples/weather/fcn_afno/README.html
6. Paper: https://arxiv.org/abs/2202.11214
## Compute Environment
I will use the GT Compute MakerSpace ICE cluster to help run inference and training down the line. Inference is light, requiring 1 GPU, while small-scale training might use 4 or 5.
